docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,624 @@
+
+```notion
+Context Precision@K = (Precision@K * Relevance of K) / 
+										  (Total number of relevant items in the top K)
+
+where
+Relevance of K =  1 for relevant / 0 for irrelevant items
+K = number of chunks
+```
+
+- [Context relevancy](https://docs.ragas.io/en/latest/concepts/metrics/context_relevancy.html) - It is computed based on the relevancy of the retrieved context with respect to the question. The value ranges between 0 and 1, with higher values indicating better relevancy. It is calculated as :
+
+```notion
+context relevancy = (Number of relevant sentences within the retrieved) / 
+										(Total number of sentences in retrieved context)
+```
+
+- [Context entity recall](https://docs.ragas.io/en/latest/concepts/metrics/context_entities_recall.html) - This metric evaluates entity retrieval based on comparison with ground truth entities to ensure relevant context coverage.The value ranges between 0 and 1, with higher values indicating better entity recall. It is calculated as :
+
+```notion
+context entity recall = (Entities in context ∩ Entities in Ground Truth) / 
+												(Entities in Ground Truth)
+```
+
+- [Answer semantic similarity](https://docs.ragas.io/en/latest/concepts/metrics/semantic_similarity.html) - This metric evaluates the semantic similarity between the generated answer and the ground truth. This evaluation is based on the ground truth and the answer, with values ranging between 0 and 1. It is calculated as :
+
+```notion
+answer similarity score = cosine similarity (Vector of ground truth , Vector of generated answer)
+```
+
+- [Answer correctness](https://docs.ragas.io/en/latest/concepts/metrics/answer_correctness.html) - This metric evaluates the accuracy between the generated answer and the ground truth. This evaluation is based on the ground truth and the answer, with values ranging between 0 and 1. It is calculated as :
+
+```notion
+answer correctness = factual correctness (ground truth , generated answer ) + answer similarity score
+
+Where factual correctness is the F1 score calculated using ground truth and generated answer.
+```
+
+
+### **Choosing the right metrics for your use cases**
+
+Ragas focuses on evaluating mainly Retrieval and Generation phases of your RAG pipeline and offers metrics to :
+
+- **Evaluate Retrieval**: *context_relevancy* and *context_recall*, which give you the fair measure of the efficiency of your retrieval system.
+- **Evaluate Generation**: *faithfulness* to measure the most important aspect when working with LLM i.e. hallucinations and *answer_relevancy* which measures the relevancy of the generated answer wrt to the question.
+
+The mean score of these 4 aspects gives you the `ragas score` which is a single measure of the performance of your RAG system on the aspects of retrieval and generation.
```

Comment:
@atarora 
Above you enumerate 8 metrics. 
Here you restate that RAGAS focuses on 4. Can you indicate what the purpose is of discussing the other 4 (as part of the 8 above)?

Reply:
4 aspects - 8 metrics 
Ref : https://docs.ragas.io/en/latest/concepts/metrics/index.html